feat: Added SPL doc for mathematical/statistical/trigonometric eval functions

Macbeth98 opened the pull request:
#### This includes SPL eval sub-commands documentation for:
- Mathematical functions
- Statistical functions
- Trigonometric and Hyperbolic functions

Netlify Link: https://spl-doc.netlify.app/

venkateshamatam commented on docusaurus.config.js:
```diff
@@ -16,7 +16,7 @@ const config = {
   // Set the /<baseUrl>/ pathname under which your site is served
   // For GitHub pages deployment, it is often '/<projectName>/'
-  baseUrl: '/siglens-docs/',
+  baseUrl: '/',
```
I don't think this change should be merged(it can be reverted after Andrew has reviewed the netlify link), it might impact the Github pages deployment.

Macbeth98 replied:
Ohh Yes, I will update it, once Andrew has reviewed.

AndrewHess replied:
I think Venky is right. But Mani, I don't quite understand; were you waiting for me to review the markdown files, or waiting for me to find if this will mess up the url/deployment?

Macbeth98 replied:
No this is just because, `Netlify` would throw an error if the baseURL is not set to `/`. 
So I believe when Venky made the initial deployment setup he might have set the url to `/` and we used Venky's `spl-docs` branch(which is the same branch deployed on Netlify). 
And I pushed that here without noticing it.

I said I would update once you have reviewed, because if I did now the Netlify link that is in PR description will not work. So I wanted to update it once you have reviewed everything and there are no further changes.

Macbeth98 replied:
I have updated this baseURL now to the original one.

AndrewHess commented on docs/spl-docs/evaluation-functions/mathematical-functions.md:
```diff
@@ -0,0 +1,441 @@
+
+This allows the user to analyze the log-transformed data to identify exponential growth patterns in website traffic.
+
+## log(\<num>, \<base>)
+
+This function returns the logarithm of a number using a base. 
+
+### Usage
+
+The `<num>` argument can be the name of a numeric field or a numeric literal.
+
+The `<base>` is optional, and if omitted the log function uses base 10.
+
```
There should be some restrictions on `num` and `base`. Mention those. Also mention what happens if those criteria are not met (does `log` give NULL?)

Macbeth98 replied:
So I have mentioned the restrictions. 
But regarding the error as of now we are throwing error on the backend, but we are not showing that on the UI.

AndrewHess commented on docs/spl-docs/evaluation-functions/mathematical-functions.md:
```diff
@@ -0,0 +1,441 @@
+
+**Calculate future investment value with compound interest**
+
+**Problem:** A user wants to calculate the future value of an investment based on compound interest. The formula for compound interest is given by `Future_Value = Principal * (1 + annual_interest_rate/times_compounded_per_year)^(times_compounded_per_year * years)`.
+
+**Solution:** The `pow()` command can be used to compute `(1 + annual_interest_rate/times_compounded_per_year)^(times_compounded_per_year * years)`.
```
nit: these equations don't display well for me. Use code blocks instead of inline code
<img width="951" alt="image" src="https://github.com/siglens/siglens-docs/assets/20426590/1e5e5e45-602b-4638-be81-678b1bcadeaf">

AndrewHess commented on docs/spl-docs/evaluation-functions/mathematical-functions.md:
```diff
@@ -0,0 +1,441 @@
+
+This ensures that the user can accurately calculate the area of a circle given its radius using the precise value of π.
+
+## sqrt(\<num>)
+
+This function returns the square root of a number.
+
+### Usage
+
+The `<num>` argument can be the name of a numeric field or a numeric literal.
+
```
What happens for negative numbers?

Macbeth98 replied:
It does not work, I will add this information.

AndrewHess commented on docs/spl-docs/evaluation-functions/mathematical-functions.md:
````diff
@@ -0,0 +1,441 @@
+... | eval n=sqrt(9)
+```
+
+### Use-Case Example
+
+**Calculate the hypotenuse of a right triangle**
````
nit: maybe change this to something a bit more practical, like calculating standard deviation of something (number of requests, latency, etc.)

Macbeth98 replied:
Calculating standard deviation requires using lot of `multi value` commands like `mvsum`, `mvmap`, `mvcount`, etc. Which I do not think we support yet. And also the command `stats values(latency)`, I do not think the output is correct.  

So for now I have used something else called `Root Mean Square Error (RMSE) of predictions`.

AndrewHess commented on docs/spl-docs/evaluation-functions/statistical-functions.md:
````diff
@@ -0,0 +1,185 @@
+server2   1720350060   89
+```
+
+## min(\<values>)
+
+This function one or more numeric or string values and returns the minimum value. Strings are greater than numbers.
+
+### Usage
+
+The `<values>` argument can be list of strings or numbers or a field name.
````
> ... can be list of strings or numbers or a field name.

Do you mean:
```
each element can be a literal string, literal number, or a field name
```

Macbeth98 replied:
Yes. I have updated the statement.

AndrewHess commented on docs/spl-docs/evaluation-functions/statistical-functions.md:
```diff
@@ -0,0 +1,185 @@
+
+## random()
+
+This function returns a pseudo-random integer ranging from `0 to 231-1`.
```
> `0 to 231-1`.

You forgot an exponent. Alternatively you could say something like:
```
... returns a pseudo-random integer between 0 and 2147483647 inclusive
```
But do whatever you think is best

AndrewHess commented on docs/spl-docs/evaluation-functions/statistical-functions.md:
```diff
@@ -0,0 +1,185 @@
+
+This example takes a random number and uses the modulo mathematical operator `( % )` to divide the random number by `100000`. This ensures that the random number returned is not greater than `100000`. The number remaining after the division is increased by `1` to ensure that the number is at least greater than or equal to `1`.
+
+### Use-Case Example
+
+**Simulate random discounts for a marketing campaign**
```
nit: I'm not sure you'd want to do this in splunk (but maybe you would). One alternative for `random()` would be if you want to do an analysis on your data for a certain time frame but you have a lot of data so it would take a long time. You could use `random()` to randomly select some small percentage of the records within that time frame and just do the analysis on that

AndrewHess commented on docs/spl-docs/evaluation-functions/trig-hyperbolic-functions.md:
````diff
@@ -0,0 +1,189 @@
+... | eval n=acos(0)
+```
+
+The following example calculates 180 divided by pi and multiplies the result by the arc cosine of 0. This example returns `90.0000000003`.
+
+```
+... | eval degrees=acos(0)*180/pi()
+```
````
You already have the `acos(0)` example, and a lot of your comment on line 17 is obvious from the code. I'd suggest either removing this example or changing the description to something more useful, such as:
```
... converts `acos(0)` from radians to degrees
```

Similar comment for your `asin` example

AndrewHess commented on docs/spl-docs/evaluation-functions/trig-hyperbolic-functions.md:
```diff
@@ -0,0 +1,189 @@
+
+The following list contains the functions that you can use to calculate trigonometric and hyperbolic values.
+
+## acos(x)
+
+This function computes the arc cosine of `x`, in the interval `[0,pi]` radians.
```
What happens if `x` is outside the domain?

Macbeth98 replied:
We are throwing an error, but that error is not shown on the UI. It just shows empty results.

AndrewHess commented on docs/spl-docs/evaluation-functions/trig-hyperbolic-functions.md:
For these functions, you should specify the domain and what happens if the input is not in the domain (NULL, or something else?)

Macbeth98 commented on docs/spl-docs/evaluation-functions/mathematical-functions.md:
```diff
@@ -0,0 +1,441 @@
+
+This ensures that the average transaction amounts are displayed with two decimal places, making the financial report clear and consistent.
+
+## ln(\<num>)
+
+This function returns the natural logarithm of a number.
```
Yes, I think it is good to clarify. But how do you suggest we do that?

AndrewHess replied:
Adding this or something similar to the description should be fine
```
Note: This is "ln" (lowercase L for natural logarithm), not "In" (uppercase I).
```

AndrewHess commented:
For the cases you mentioned where invalid inputs gives no results in the UI and an error in the backend, I think saying `the behavior is undefined` is fine, unless you want to give users more detail

Macbeth98 commented:
> For the cases you mentioned where invalid inputs gives no results in the UI and an error in the backend, I think saying `the behavior is undefined` is fine, unless you want to give users more detail

Got it. Is there anything else that I should address apart from these?

AndrewHess commented:
> Got it. Is there anything else that I should address apart from these?

I think it's just those and the `ln` clarification. Then this looks good to merge

Actually, there's also Venky's comment and I'm not sure what needs to be done. Are you going to revert that, or do we want to keep that change?

Macbeth98 commented:
> > Got it. Is there anything else that I should address apart from these?
> 
> I think it's just those and the `ln` clarification. Then this looks good to merge
> 
> Actually, there's also Venky's comment and I'm not sure what needs to be done. Are you going to revert that, or do we want to keep that change?

Hey Andrew, I believe I have addressed all the comments. And I think this is good to merge now.
